### DISCUSSION TITLE
How to attach a volume just to specific nodes?

### DISCUSSION THREAD

**Discussion by liviu-macsen** [longhorn/longhorn#3574] [Q&A]
Hello,

In our Custer we have multiple nodes from 3 different regions (let's say A, B and C).
Each node has tags with zone name. When I create a volume for region A and mount it to a workload, replicas are made on A region nodes, but the problem is that it become attached to a random server. And most of the time is attached to a node from region B or C. and performance is very poor because bandwidth between region is weak. 
Is there a method to explicit attach a volume to some specific nodes in Longhorn, or I miss something?

Thank you!

**Comment by derekbit** [comment]
You can move your workload using the volume to region A.

**Comment by liviu-macsen** [comment]
The workload and replicas are in right region, but the problem si when the volume is attached to a node from another region.
It seems that traffic is routed through serverB-node.  

<img width="1314" alt="Screenshot 2022-02-01 at 09 45 12" src="https://user-images.githubusercontent.com/80744465/151930650-1970a4ac-67b4-4b1d-be88-8deb3452a049.png">

**Comment by kevinono62** [comment]
Hello, I have the same issue with data locality in 'best effort' mode not helping. When a pod attaches to a volume located in a different zone than the pod itself, I experience application latencies. I would like the pod to attach to the volume within the same zone.

**Comment by ChanYiLin** [comment]
Hi @kevinono62 
The volume should be attached to the same node as workload pod.
Are you saying that your volume is attached to NodeA and the pod is running on NodeB?
Is this a RWX Volume?

**Reply by R0flcopt3r** [reply]
I have an RWX volume, datalocality to best effort, and node/disk tags that make sure the volume can only live on one specific node, and it's very fast disk. The attached node seems to be completely random. Sometimes it lands on a node with very slow networking. The way i read this is the files are then sent from the node hosting it, to the node attaching it, and then back to whatever node holds the workload. Seems very inefficient. I'm seeing better performance on our ancient 10Gb Synology NAS using NFS..

**Reply by derekbit** [reply]
@R0flcopt3r We have introduced the data locality feature for RWX volume in v1.7.0. Please check https://longhorn.io/docs/1.7.0/nodes-and-volumes/volumes/rwx-volumes/#configuring-volume-locality-for-rwx-volumes.
